test/python/compiler/test_assembler.py

```diff
@@ -385,13 +385,13 @@ def setUp(self):
     def test_assemble_sample_pulse(self):
         """Test that the pulse lib and qobj instruction can be paired up."""
         schedule = pulse.Schedule()
-        schedule += pulse.Play(pulse.SamplePulse([0.1]*16, name='test0'),
+        schedule += pulse.Play(pulse.Waveform([0.1]*16, name='test0'),
                                pulse.DriveChannel(0),
                                name='test1')
-        schedule += pulse.Play(pulse.SamplePulse([0.1]*16, name='test1'),
+        schedule += pulse.Play(pulse.Waveform([0.1]*16, name='test1'),
                                pulse.DriveChannel(0),
                                name='test2')
```

Comment:
We still have to test that `SamplePulse` can be assembled, because it's still supported for a while.
```suggestion
        schedule += pulse.Play(pulse.SamplePulse([0.1]*16, name='test1'),
                               pulse.DriveChannel(0),
                               name='test2')
```

Reply:
Perhaps a better alternative would be to keep this whole test how it was, make a comment that it's deprecated, and then add another test called `test_assemble_waveform` that contains this exact code. If you only commit the above suggestion, you might want to rename this test to `test_assemble_waveform`